src/infrastructure/jira/jira-service.ts

```diff
@@ -173,6 +171,100 @@ class JiraService {
 			}
 		}
 	};
+
+	deleteDesignUrlInIssueProperties = async (
+		issueIdOrKey: string,
+		design: AtlassianDesign,
+		connectInstallation: ConnectInstallation,
+	): Promise<void> => {
+		await Promise.all([
+			await this.deleteAttachedDesignUrlInIssuePropertiesIfPresent(
+				issueIdOrKey,
+				design,
+				connectInstallation,
+			),
+			await this.deleteFromAttachedDesignUrlV2IssueProperties(
+				issueIdOrKey,
+				design,
+				connectInstallation,
+			),
+		]);
+	};
+
+	/**
+	 * @internal
+	 * Only visible for testing. Please use {@link deleteDesignUrlInIssueProperties}
+	 */
+	deleteAttachedDesignUrlInIssuePropertiesIfPresent = async (
+		issueIdOrKey: string,
+		design: AtlassianDesign,
+		connectInstallation: ConnectInstallation,
+	): Promise<void> => {
+		try {
+			const response = await jiraClient.getIssueProperty(
```

Comment:
**nitpick:** It seems we could attempt to delete the Issue Property straightaway without fetching it first. Both `GET` and `DELETE` endpoints throw `JiraClientNotFoundError` if the Issue Property is not found.

Reply:
No we can't do this since a user may link more than 1 design to the issue, and if you recall, the way the v1 property works (`attached-design-url`) is that the **first** linked design is stored. If further designs are linked, the v1 property does not update. So, only when the initially linked design is removed should we do the delete.

Example:
- A user creates an issue, and links design 1 as their first linked design
- `attached-design-url` _is set to design 1_
- User links design 2 to the issue
- `attached-design-url` _is not updated_
- User removes design 2 from the issue
- `attached-design-url` _is not deleted_
- User removes design 1 from the issue
- `attached-design-url` _is deleted_

Reply:
Oh, I see! I've overlooked this. Thank you for the explanation, Costa!